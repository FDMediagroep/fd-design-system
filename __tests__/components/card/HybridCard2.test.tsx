import React from 'react';
import {
    HybridCard2,
    getCssClassNames,
} from '../../../src/components/card/HybridCard2';
import { render } from '@testing-library/react';

describe('HybridCard2', () => {
    test('should render correctly', () => {
        const { container, getByText } = render(
            <HybridCard2
                id="1"
                label="Ondernemen"
                title="This is the title"
                url="https://example.com/1/This is the title"
            />
        );

        expect(getByText('This is the title')).toBeTruthy();
        expect(getByText('Ondernemen')).toBeTruthy();
        expect(container.querySelector('[href]').getAttribute('href')).toBe(
            'https://example.com/1/This is the title'
        );
    });

    test('should render is-read correctly', () => {
        const { container } = render(
            <HybridCard2
                id="1"
                label="Ondernemen"
                title="This is the title"
                imageUrl="https://images.example.com/image.png"
                imageUrlS="https://images.example.com/imageS.png"
                imageUrlM="https://images.example.com/imageM.png"
                imageUrlL="https://images.example.com/imageL.png"
                url="https://example.com/1/This is the title"
                isRead={true}
            />
        );
        expect(container).toMatchSnapshot();
    });

    test('should render variant-1 correctly', () => {
        const { container } = render(
            <HybridCard2
                id="1"
                label="Ondernemen"
                title="This is the title"
                imageUrl="https://images.example.com/image.png"
                imageUrlS="https://images.example.com/imageS.png"
                imageUrlM="https://images.example.com/imageM.png"
                imageUrlL="https://images.example.com/imageL.png"
                url="https://example.com/1/This is the title"
                variant="variant-1"
            />
        );
        expect(container).toMatchSnapshot();
    });

    test('should have correct class name', () => {
        const { container } = render(
            <HybridCard2
                id="1"
                label="Ondernemen"
                title="This is the title"
                imageUrl="https://images.example.com/image.png"
                imageUrlS="https://images.example.com/imageS.png"
                imageUrlM="https://images.example.com/imageM.png"
                imageUrlL="https://images.example.com/imageL.png"
                url="https://example.com/1/This is the title"
            />
        );

        expect(
            container.firstElementChild.classList.contains(
                getCssClassNames()[0]
            )
        ).toBeTruthy();
    });
});
