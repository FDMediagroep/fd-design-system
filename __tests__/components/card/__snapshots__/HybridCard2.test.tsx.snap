// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HybridCard2 should render is-read correctly 1`] = `
<div>
  <article
    class="hybridCard2 isRead"
    id="1"
  >
    <a
      aria-label="This is the title"
      class="xs__p+4"
      href="https://example.com/1/This is the title"
    >
      <figure
        class="xs__m-0 s__mt-4 s__mr-4 s__mb+3 s__ml-4"
      >
        <picture>
          <source
            media="(max-width:639px)"
            srcset="https://images.example.com/imageS.png"
          />
          <source
            media="(max-width:860px)"
            srcset="https://images.example.com/imageM.png"
          />
          <source
            media="(min-width:861px)"
            srcset="https://images.example.com/imageL.png"
          />
          <img
            alt="This is the title"
            src="https://images.example.com/image.png"
          />
        </picture>
        <figcaption />
      </figure>
      <div
        class="teaserText xs__pl+4 s__pr-0 s__pb-0 s__pl-0"
      >
        <div
          class="meta xs__mb+2 body-text sans s"
        >
          <span
            class="prefix"
          >
            Ondernemen
          </span>
        </div>
        <div
          class="textContainer"
        >
          <h1
            class="heading serif xs xs__m-0"
          >
            This is the title
          </h1>
        </div>
      </div>
    </a>
  </article>
</div>
`;

exports[`HybridCard2 should render variant-1 correctly 1`] = `
<div>
  <article
    class="hybridCard2"
    id="1"
  >
    <a
      aria-label="This is the title"
      class="xs__p+4"
      href="https://example.com/1/This is the title"
    >
      <figure
        class="xs__m-0 s__mt-4 s__mr-4 s__mb+3 s__ml-4"
      >
        <picture>
          <source
            media="(max-width:639px)"
            srcset="https://images.example.com/imageS.png"
          />
          <source
            media="(max-width:860px)"
            srcset="https://images.example.com/imageM.png"
          />
          <source
            media="(min-width:861px)"
            srcset="https://images.example.com/imageL.png"
          />
          <img
            alt="This is the title"
            src="https://images.example.com/image.png"
          />
        </picture>
        <figcaption />
      </figure>
      <div
        class="teaserText xs__pl+4 s__pr-0 s__pb-0 s__pl-0"
      >
        <div
          class="meta xs__mb+2 body-text sans s"
        >
          <span
            class="prefix"
          >
            Ondernemen
          </span>
        </div>
        <div
          class="textContainer"
        >
          <h1
            class="heading serif xs xs__m-0"
          >
            This is the title
          </h1>
        </div>
      </div>
    </a>
  </article>
</div>
`;
